CMakeLists.txt

```diff
@@ -34,6 +34,7 @@ endforeach()
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_ENABLE_EXPORTS ON)
```

Comment:
as I understand it after reading https://cmake.org/cmake/help/latest/prop_tgt/ENABLE_EXPORTS.html, `CMAKE_ENABLE_EXPORTS` only affects executables. is that correct?